ethereum_clients/Plugin.js

```diff
@@ -182,7 +183,37 @@ class Plugin extends EventEmitter {
     return undefined
   }
 
-  async start(release, flags) {
+  async requestStart(app, flags, release) {
+    return new Promise((resolve, reject) => {
+      // TODO move dialog code to different module
+      dialog.showMessageBox(
+        // currentWindow,
+        {
+          title: 'Start requested',
+          buttons: ['OK', 'Cancel'],
+          message: `The application "${
+            app.name
+          }" requests to start the client or service "${
+            this.displayName
+          }" with flags: [${
+            flags ? flags.join(' ') : ''
+          }].\n\nPress 'OK' to allow this time.
+        `
+        },
+        async response => {
+          const userPermission = response !== 1 // = index of 'cancel'
+          if (userPermission) {
+            await this.start(flags, release)
+            resolve()
+          } else {
+            reject(new Error('user permission denied'))
```

Comment:
strictly speaking, this should not generate an error message, as it's not an error. we can handle it better another time.

![image](https://user-images.githubusercontent.com/47108/61743258-e876a280-ad62-11e9-85a2-c1634160c932.png)